src/MongoDB.Driver.Core/Core/Operations/FindAndModifyOperationBase.cs

```diff
@@ -208,15 +208,6 @@ private TResult ProcessCommandResult(ConnectionId connectionId, RawBsonDocument
                 GuidRepresentation = _messageEncoderSettings.GetOrDefault<GuidRepresentation>(MessageEncoderSettingsName.GuidRepresentation, GuidRepresentation.CSharpLegacy)
             };
 
-            BsonValue writeConcernError;
-            if (rawBsonDocument.TryGetValue("writeConcernError", out writeConcernError))
-            {
-                var message = writeConcernError["errmsg"].AsString;
-                var response = rawBsonDocument.Materialize(binaryReaderSettings);
-                var writeConcernResult = new WriteConcernResult(response);
-                throw new MongoWriteConcernException(connectionId, message, writeConcernResult);
-            }
-
```

Comment:
I suspect that now that we're no longer inspecting the result for a writeConcernError that the helper WriteCommandOperation could return a TResult directly instead of a RawBsonDocument but I chose to make the smallest change possible.